ocollection.h

```diff
@@ -0,0 +1,90 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+struct OVector{
```

Comment:
a better way to define a struct is to use `typedef struct _OVector{...} OVector;`

after this change u can pass your structure as function parameter without `struct` keyword